connection.go

```diff
@@ -239,6 +239,16 @@ func (c *Connection) WithContext(ctx context.Context) *Connection {
 	return cn
 }
 
+// WithNowFuncContext returns a copy of the connection, wrapped with a context containing the custom `nowFunc`.
+func (c *Connection) WithNowFuncContext(ctx context.Context, nowFunc func() time.Time) *Connection {
```

Comment:
This will break if you do
```go
c = c.WithNowFuncContext(...)
// ...
c.WithContext(ctx).All(...)
```
If we modify the connection anyway, why bother with the context at all? We can just set a scoped now func on the connection directly?